src/megaapi_impl.cpp

```diff
@@ -23127,6 +23669,116 @@ void MegaSyncPrivate::setRegExp(MegaRegExp *regExp)
     }
 }
 
+int MegaSyncPrivate::getError() const
+{
+    return mError;
+}
+
+void MegaSyncPrivate::setError(int error)
+{
+    mError = error;
+}
+
+void MegaSyncPrivate::disable(int error)
+{
+    setState(SYNC_DISABLED);
+    setError(error);
+}
+
+bool MegaSyncPrivate::isEnabled() const
+{
+    return state != SYNC_CANCELED && (state != SYNC_DISABLED || mError != NO_SYNC_ERROR );
+}
+
+bool MegaSyncPrivate::isActive() const
+{
+    return state != SYNC_FAILED && state != SYNC_CANCELED && state != SYNC_DISABLED;
+}
+
+bool MegaSyncPrivate::isTemporaryDisabled() const
+{
+    return state == SYNC_DISABLED && mError != NO_SYNC_ERROR;
+}
+
+
+MegaSyncListPrivate::MegaSyncListPrivate()
+{
+    list = NULL;
+    s = 0;
+}
+
+MegaSyncListPrivate::MegaSyncListPrivate(MegaSyncPrivate** newlist, int size)
+{
+    list = NULL; s = size;
+    if(!size) return;
+
+    list = new MegaSync*[size];
```

Comment:
I strongly recommend to use a STL container and maybe smart pointers to handle the items stored by the `MegaSyncListPrivate`. I'll only review the implementation of this class after decision to not change (or after changes).

Reply:
MegaSyncListPrivate is an exact copy of MegaNodeListPrivate. I'd rather keep the same code. But I can change it if you want.

Reply:
Already decided not to change it :)